pkg/ruler/api.go

```diff
@@ -0,0 +1,264 @@
+package ruler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/weaveworks/common/user"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+// RegisterRoutes registers the configs API HTTP routes with the provided Router.
+func (r *Ruler) RegisterRoutes(router *mux.Router) {
+	if r.store == nil {
+		level.Info(util.Logger).Log("msg", "ruler configured with store that does not support api")
+		return
+	}
+	for _, route := range []struct {
+		name, method, path string
+		handler            http.HandlerFunc
+	}{
+		{"get_rules", "GET", "/api/prom/api/v1/rules", r.rules},
```

Comment:
This doesn't seem to be a universal pattern throughout cortex. For instance:
https://github.com/cortexproject/cortex/blob/dd77369a2341b2ea263ecf8cd01fae2d347cbd16/pkg/cortex/modules.go#L258
https://github.com/cortexproject/cortex/blob/dd77369a2341b2ea263ecf8cd01fae2d347cbd16/pkg/cortex/modules.go#L261

I am in favor of standardizing around `cortex.Config.HTTPPrefix`. This relates to the concerns I brought up in #1722

Reply:
Thanks for pointing #1722. I've commented there. Given I do agree this is **not** the right PR to also address the `cortex.Config.HTTPPrefix` standardization, and given you've correctly pointed out that there are more places where it's hardcoded than based on `HTTPPrefix`, I'm going to resolve this comment, so that we can continue this discussion in #1722 before working on it in a separate PR.